drivers/i2c/hts221.py

```diff
@@ -0,0 +1,101 @@
+"""
+HTS221 - STMicro I2C temperature and humidity sensor driver for MicroPython
+Example usage:
+>>> from hts221 import HTS221
+>>> hts = HTS221(I2C(1))
+>>> hts.measure()
+>>> hts.get_humidity()
+62.07
+>>> hts.get_temperature()
+27.046
+"""
+import array
+import uctypes
+import time
+
+class HTS221:    
+    HTS_WHO_AM_I = const(0xf)
+    
+    def __init__(self, i2c, addr=95, fixed=False):
+        self.i2c = i2c
+        self.addr = addr
+        self.fixed = fixed
+        if (self.i2c.readfrom_mem(addr, HTS_WHO_AM_I, 1) != b'\xbc'):
+            raise OSError("No HTS221 device on address {} found".format(addr))
+        self.cal_buf = bytearray(16)
+        self.calib = uctypes.struct(uctypes.addressof(self.cal_buf), {
+                'h0_rh':(uctypes.UINT8 | 0), 
+                'h1_rh':(uctypes.UINT8 | 1), 
+                't0_deg':(uctypes.UINT8 | 2), 
+                't1_deg':(uctypes.UINT8 | 3), 
+                'res':(uctypes.UINT8 | 4),
+                't1t0msb':(uctypes.UINT8 | 5),
+                'h0_out':(uctypes.INT16 | 6),
+                'res1':(uctypes.INT16 | 8),
+                'h1_out':(uctypes.INT16 | 10),
+                't0_out':(uctypes.INT16 | 12),
+                't1_out':(uctypes.INT16 | 14)})
+
+        self.val_buf = bytearray(5)
+        self.raw_val = uctypes.struct(uctypes.addressof(self.val_buf), {
+                'status':(uctypes.UINT8 | 0), 
+                'h_out':(uctypes.INT16 | 1), 
+                't_out':(uctypes.INT16 | 3)})
+        
+        # enable sensor, one-shoot
+        i2c.writeto_mem(addr, 0x20, b'\x84')
+        i2c.readfrom_mem_into(95, 0x30|0x80, self.cal_buf)
+        
+        # add 2 bits from reg 0x35 -> 10bit unsigned values)
+        self.t0_deg = (self.calib.t0_deg | ((self.calib.t1t0msb & 0x3) << 8))
+        self.t1_deg = (self.calib.t1_deg | ((self.calib.t1t0msb & 0xc) << 6))
+        
+        # multiply with large base to keep it accurate for fixed point division
+        self.t_slope = (1000 * (self.t1_deg - self.t0_deg)) // (self.calib.t1_out - self.calib.t0_out)
+        self.h_slope = (10000 * (self.calib.h1_rh - self.calib.h0_rh)) // (self.calib.h1_out - self.calib.h0_out)
+        
+    def measure(self):
+        # enable one-shot measurement
+        self.i2c.writeto_mem(self.addr, 0x21, b'\x01')
+        self.measure_done = False
+
+    def measure_end(self):
+        # blocking, timeout after 2sec
```

Comment:
thanks for pointing out - yes it's not correct. It should be 0.2sec.

Actually the datasheet is a bit unclear about the sampling time in one-shot mode, but I've measured it to be always smaller than 10msec.